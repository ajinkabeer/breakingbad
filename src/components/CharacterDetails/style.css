img {
  border-radius: 8;
}

p {
  text-align: center;
}

.content-container {
  display: 'flex';
  flex-direction: 'column';
  justify-content: 'center';
  align-items: 'center';
}
