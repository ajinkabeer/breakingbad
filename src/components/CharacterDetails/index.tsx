import './style.css';

interface Props {
  img: string;
  name: string;
  nickname: string;
  birthday: string;
  portrayed: string;
  category: string;
}

const CharacterDetails = ({
  img,
  name,
  nickname,
  birthday,
  portrayed,
  category,
}: Props) => (
  <div>
    <div className="content-container">
      <img src={img} height={400} alt={name} />
    </div>
    <div className="content-container">
      <p>
        Name: <strong>{name}</strong>
      </p>
      <p>
        Nickname: <strong>{nickname}</strong>
      </p>
      <p>
        Birthday: <strong>{birthday}</strong>
      </p>
      <p>
        Portrayed by: <strong>{portrayed}</strong>
      </p>
      <p>
        Category: <strong>{category}</strong>
      </p>
    </div>
  </div>
);

export default CharacterDetails;
