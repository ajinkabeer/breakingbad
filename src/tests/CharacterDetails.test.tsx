import { render } from '@testing-library/react';
import CharacterDetails from '../components/CharacterDetails';

const props = {
    name: 'testName',
    img: 'test.png',
    nickname: 'testnickname',
    birthday: '18-01-1995',
    portrayed: 'testactor',
    category: 'testcategory'
};
const {
    img,
    name,
    nickname,
    birthday,
    portrayed,
    category
} = props;

describe('Character Component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <CharacterDetails
                img={img}
                name={name}
                nickname={nickname}
                birthday={birthday}
                portrayed={portrayed}
                category={category}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders the correct props', () => {
        const { getByText } = render(
            <CharacterDetails
                img={img}
                name={name}
                nickname={nickname}
                birthday={birthday}
                portrayed={portrayed}
                category={category}
            />
        );
        getByText('testName');
        getByText('testnickname');
        getByText('18-01-1995');
        getByText('testactor');
        getByText('testcategory');
    });
});
