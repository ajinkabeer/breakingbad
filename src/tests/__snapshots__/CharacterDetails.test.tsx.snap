// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="content-container"
    >
      <img
        alt="testName"
        height="400"
        src="test.png"
      />
    </div>
    <div
      class="content-container"
    >
      <p>
        Name: 
        <strong>
          testName
        </strong>
      </p>
      <p>
        Nickname: 
        <strong>
          testnickname
        </strong>
      </p>
      <p>
        Birthday: 
        <strong>
          18-01-1995
        </strong>
      </p>
      <p>
        Portrayed by: 
        <strong>
          testactor
        </strong>
      </p>
      <p>
        Category: 
        <strong>
          testcategory
        </strong>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
